internal/gitinterface/utils.go

```diff
@@ -37,3 +38,35 @@ func RemoteRef(refName, remoteName string) string {
 
 	return remotePath
 }
+
+// IsNiceGitVersion determines whether the version of git is "nice". Certain Git
+// subcommands that gittuf uses were added in newer versions than some common
+// client versions. Instead of using a workaround for all clients, we determine
+// if we can use the newer features or instead need to use workarounds.
+func IsNiceGitVersion() (bool, error) {
```

Comment:
I think this can be private. Also, we can leave this be by now but if we encounter other cases where we have different minimum version constraints, we could have the constraint as a parameter here. Let's assess if / when we need it though.